### PR TITLE
[go-gen] Add setup.go and invoke from main function

### PR DESCRIPTION
**Pull request by jaylees14**
This PR does 2 main things:
- Add `setup.go`, which includes a `setup` function for users to add their own hooks / logic / routes etc.
- Refactor the router so it is no longer defined on the environment, instead it is returned and passed to the user for modification

### Test Plan
- Existing integration tests cover that the code complies and endpoints still produce correct responses

### CODE REVIEW

File: src/e2e/resources/simple-temple-expected/auth/auth.go [hunk 1]
```diff
@@ -48,8 +48,8 @@ type loginAuthResponse struct {
 	AccessToken string
 }
 
-// router generates a router for this service
-func (env *env) router() *mux.Router {
+// defaultRouter generates a router for this service
+func defaultRouter(env *env) *mux.Router {
```
Comment (c-harding) [inline src/e2e/resources/simple-temple-expected/auth/auth.go:52]: When would a router not be a default router?

**Review by c-harding — APPROVED** [review]
All the power to Captain Hook!

**Review by Lewiky — APPROVED** [review]
So many changes for such little extra logic

File: src/e2e/resources/simple-temple-expected/auth/auth.go (hunk 1 above)
Comment (jaylees14) [reply src/e2e/resources/simple-temple-expected/auth/auth.go:52]: The router gets given to the user for modification, this just defaults to everything we've generated

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀 

The `Call into non-generated entry-point` comment seems a bit funny, since the `env.setup` could theoretically do more than add an entry point?

**Pull request merged**